docs/development/new-cloud-provider.md

```diff
@@ -0,0 +1,56 @@
+# Adding Cloud Providers
+This document provides an overview of how to integrate a new cloud provider into Gardener. Each component that requires integration has a detailed description of how to integrate it and the steps required.
+
+## Cloud Components
+Gardener is composed of 2 or more Kubernetes clusters:
+
+* Shoot: These are one more "workload" clusters, the regular Kubernetes clusters you have seen. They provide places for your workloads to run.
+* Seed: This is the "management" cluster. It provides management service for one or more Shoot cluster.
+
+These two clusters can run in the same cloud provider, but they do not need to. For example, you could run your Seed in AWS, while having one shoot in Azure, two in Google, two in Alicloud, and three in Packet.
+
+The Seed cluster deploys and manages the Shoot clusters. Importantly, for this discussion, the `etcd` data store backing each Shoot runs as workloads inside the Seed. Thus, to use the above example, the clusters in Azure, Google, Alicloud and Packet will have their worker nodes and master nodes running in those clouds, but the `etcd` clusters backing them will run as separate [deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) in the Seed Kubernetes cluster on AWS.
+
+This distinction becomes important when preparing to integrate a new cloud provider.
+
+## Gardener Cloud Integration
+Gardener integrates with cloud providers at the following key lifecycle elements:
+
+* Create/destroy/get/list machines for the Shoot
+* Create/destroy/get/list machines for the Seed
+* Backup/restore etcd for the Seed via writing files to and reading them from object storage
+
+Thus, the integrations you need for your cloud provider depend on whether you want to deploy Shoot clusters to the provider, Seed or both.
+
+* Shoot Only: machine lifecycle management
+* Seed: machine lifecycle management **and** etcd backup/restore
```

Comment:
not only those two, but also configuration of the control plane components

Reply:
Same like above - take a look at the CloudBotanist interface. The K8s control plane (apiserver, controllers, scheduler, etc.) might need specific flags based on the cloud provider they serve. Usually not much, but there might be the one or the other special flag.